webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslationService.java

```diff
@@ -72,9 +68,30 @@ public void createPendingMTEntitiesInBatches(Long repositoryId, Set<Long> tmText
     }
   }
 
-  protected void sendForDeletion(TmTextUnitPendingMT pendingMT) {
-    logger.debug("Sending pending MT for deletion: {}", pendingMT);
-    pendingMTDeletionSink.tryEmitNext(pendingMT);
+  @Transactional
+  public void updateVariantStatusToMTReview(List<Long> currentVariantIds) {
```

Comment:
How large batch sizes / SQL commands do we expect this to support in practice? (there is always a limit to SQL query length)

In Push/PullRun service JDBC operations, we define batch sizes for the JDBC operations and split them before issuing the SQL commands (IIRC, we use a batch size of 1000) - we should enforce a size here as well.

Reply:
In practice the max size of this id list will be 5004, as that's the maximum size per batch file sent to Smartling so in this scenario all of the text units in the pushed file have a AI translation.

I will update this function to split the passed list into batches of configurable size prior to the sql command being issues